### DISCUSSION TITLE
Clarification on Route Mappings for Middleware in Knowlg Services

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#219] [Q&A] [answered]
Hello team,

We have recently set up Knowlg and inQuiry services and are looking to integrate the generic editor (upload editor) into our application. Since we don't have a predefined middleware for these services, we're developing our own.

While testing the generic editor from the Sunbird portal, we noticed the following API calls being triggered in the network tab:

```
{{host}}/action/content/v3/create
{{host}}/action/content/v3/upload/url/do_id
{{host}}/action/content/v3/upload/do_id
{{host}}/action/content/v3/read/do_id?mode=edit&fields=body,collaborators,editorState
{{host}}/action/content/v3/update/do_id
{{host}}/action/content/v3/review/do_id
{{host}}/api/content/v1/read/do_id?fields=transcripts,ageGroup,appIcon,artifactUrl
{{host}}/content/content/v1/reject/do_id
{{host}}/content/content/v1/publish/do_id

```
We are finding it challenging to understand how these `/action/content/v3/*` or `/action/content/v1/*` endpoints map to the routes in the content service. For reference, we are using the routes defined here:
https://github.com/Sunbird-Knowlg/knowledge-platform/blob/release-6.0.0/content-api/content-service/conf/routes

In the routes file, we see both `/content/v3/*` and `/content/v4/*` paths are available. However, since the endpoints from the generic-editor are predefined and we don't have control over them, we're unsure about a few things:

Should our middleware point `/action/content/v3/*` routes to `/content/v3/*` or `/content/v4/* ` in the backend service?
Additionally, where should `/action/content/v1/* `routes point to in the middleware?
We couldn't find a direct mapping for `/content/v1` in the routes file.

We would appreciate any guidance on how to properly configure our middleware to handle these requests.

Thank you for your help!

@pallakartheekreddy @HarishGangula @rajeevsathish @vinukumar-vs @vaibhavbhuva @krgauraw @amitpriyadarshi
@coolbung @itsvick @snehal0904 @arati-tekdi @rushi-tekdi

**Comment by pallakartheekreddy** [comment]
@rajnishdargan Which version of Knowlg you are using?

**Reply by rajnishdargan** [reply]
I am not aware of the version; we installed the services using the Sunbird installer from release-7.0.0: https://github.com/project-sunbird/sunbird-ed-installer/tree/release-7.0.0.

I assume it is not an old version, and both the /v3 and /v4 APIs work when we try to consume the service without middleware. For implementing the middleware, I would like to know whether we should use /v3 or /v4.

**Reply by pallakartheekreddy** [reply, marked as answer]
Please use `/v4`
